### PR TITLE
Implement check and add headers in mage

### PR DESCRIPTION
**Pull request by jsoriano**
Add a framework to wrap go tooling in mage, [taken from go-txtfile](https://github.com/elastic/go-txfile/tree/master/dev-tools/lib/mage/gotool).

Using it, reimplements all uses of go-licenser in mage.
go-licenser is vendored.

Replaces #13211, see comments there for more context.

### CODE REVIEW

File: dev-tools/mage/gotool/go.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (jsoriano) [inline dev-tools/mage/gotool/go.go]: This file is verbatim copied from https://github.com/elastic/go-txfile/tree/master/dev-tools/lib/mage/gotool

**Comment by jsoriano** [comment]
jenkins, test this again please

File: auditbeat/magefile.go [hunk 2]
```diff
@@ -205,3 +205,13 @@ func PythonIntegTest(ctx context.Context) error {
 		return devtools.PythonNoseTest(devtools.DefaultPythonTestIntegrationArgs())
 	})
 }
+
+// AddLicenseHeaders adds license headers
```
Comment (andrewkroh) [inline auditbeat/magefile.go]:
```suggestion
// AddLicenseHeaders adds license headers.
```

File: auditbeat/magefile.go [hunk 3]
```diff
@@ -205,3 +205,13 @@ func PythonIntegTest(ctx context.Context) error {
...
+func AddLicenseHeaders() {
+	mg.Deps(devtools.AddLicenseHeaders)
+}
+
+// CheckLicenseHeaders checks license headers
```
Comment (andrewkroh) [inline auditbeat/magefile.go]:
```suggestion
// CheckLicenseHeaders checks license headers.
```

**Comment by jsoriano** [comment]
I will continue with this after #13239

**Pull request merged**